## Q1. Descriptive Statistics on a Numeric Feature

---

### 🔍 এই Question-এ কী চাওয়া হয়েছে?

`daily_screen_time_min` column-এর উপর `.describe()` run করে তার **count, mean, std, min, max** দেখতে হবে। তারপর **Q1.2**-এ 2-3 বাক্যে comment করতে হবে।

### 🎯 এখান থেকে কী অর্জন করব?

একটি numeric column-এর **central tendency** (mean) ও **spread** (std) দেখে বুঝতে পারব data কতটা ছড়িয়ে আছে এবং কোনো extreme মান (outlier) আছে কিনা।

### 🧠 কীভাবে চিন্তা করতে হবে?

- **mean vs max দূরত্ব বড়?** → outlier থাকতে পারে
- **std বড়?** → data অনেক ছড়িয়ে আছে
- **min কি 0 বা negative?** → data-র nature বুঝতে সাহায্য করে


### Q1.1 — Numeric Column Select ও `.describe()` Run করা

In [1]:
import pandas as pd
import numpy as np

# Now we'll load the dataset and select the numeric column
df = pd.read_csv("02. Provided - module123_data.csv")
num_col = "daily_screen_time_min"
# Here we've loaded the dataset and chosen our target column.

# Now we'll run describe() to get summary statistics
df[num_col].describe()
# Here we've generated count, mean, std, min, 25%, 50%, 75%, max for the column.


count    100.000000
mean     181.890000
std       68.886951
min       60.000000
25%      122.000000
50%      178.000000
75%      243.750000
max      299.000000
Name: daily_screen_time_min, dtype: float64

`df[num_col].describe()` → একটি numeric column-এর সব গুরুত্বপূর্ণ statistics একসাথে দেখায়:
- **count** → কতটি non-null মান আছে (এখানে 100)
- **mean** → গড় মান (181.89 মিনিট)
- **std** → standard deviation — data কতটা ছড়িয়ে আছে (68.89 মিনিট)
- **min / max** → সর্বনিম্ন (60) ও সর্বোচ্চ (299) মান
- **25%, 50%, 75%** → quartile — data-র distribution বোঝায়


### Q1.2 — Short Answer [5 Marks]

> **Question:** Look at the count, mean, min, max, and standard deviation for your chosen column. In 2 to 3 sentences, comment on what you see. For example, does the max look very far from the mean, or does it look quite close?

---

**Answer:**

> The `daily_screen_time_min` column averages around 182 minutes, with a standard deviation of roughly 69 minutes — so there's a pretty wide spread in how long people are actually staring at their screens each day. The max clocks in at 299 minutes, which sits about 1.7 standard deviations above the mean. That's elevated, sure, but it's not some wild outlier. On the other end, even the lightest users are hitting at least 60 minutes, which honestly says something on its own. All told, the range runs from 60 to 299 minutes, and that gap tells you you're dealing with a genuinely mixed group when it comes to screen habits.

> `daily_screen_time_min` column-এর গড় প্রায় ১৮২ মিনিট, আর standard deviation প্রায় ৬৯ মিনিট — অর্থাৎ মানুষ প্রতিদিন কতক্ষণ screen-এর দিকে তাকিয়ে থাকে সেটায় বেশ ভালোই তারতম্য আছে। সর্বোচ্চ মান ২৯৯ মিনিট, যা গড়ের চেয়ে প্রায় ১.৭ standard deviation বেশি। এটা একটু বেশি বটে, তবে কোনো অস্বাভাবিক outlier বলা যাবে না। অন্যদিকে, সবচেয়ে কম ব্যবহারকারীরাও অন্তত ৬০ মিনিট screen দেখছে — এটা নিজেই একটা কথা বলে। সব মিলিয়ে, range হলো ৬০ থেকে ২৯৯ মিনিট, আর এই ব্যবধানটাই বলে দেয় যে screen ব্যবহারের অভ্যাসের দিক থেকে এখানে সত্যিকার অর্থেই একটি মিশ্র গোষ্ঠীর সাথে কাজ করছ।


---